homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -1,92 +1,203 @@
+class TotalConnectZoneSecurityBinarySensor(TotalConnectZoneBinarySensor):
+    """Represent an TotalConnect security zone."""
 
     def update(self):
         """Return the state of the device."""
-        self._is_tampered = self._zone.is_tampered()
-        self._is_low_battery = self._zone.is_low_battery()
-
         if self._zone.is_faulted() or self._zone.is_triggered():
-            self._is_on = True
+            self._attr_is_on = True
         else:
-            self._is_on = False
-
-    @property
-    def is_on(self):
-        """Return true if the binary sensor is on."""
-        return self._is_on
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from BinarySensorDeviceClass."""
-        if self._zone.is_type_security():
-            return BinarySensorDeviceClass.DOOR
-        if self._zone.is_type_fire():
-            return BinarySensorDeviceClass.SMOKE
-        if self._zone.is_type_carbon_monoxide():
-            return BinarySensorDeviceClass.GAS
-        if self._zone.is_type_motion():
-            return BinarySensorDeviceClass.MOTION
-        if self._zone.is_type_medical():
-            return BinarySensorDeviceClass.SAFETY
-        return None
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        attributes = {
-            "zone_id": self._zone_id,
-            "location_id": self._location_id,
-            "low_battery": self._is_low_battery,
-            "tampered": self._is_tampered,
-            "partition": self._zone.partition,
+            self._attr_is_on = False
+
+
+class TotalConnectLowBatteryBinarySensor(TotalConnectZoneBinarySensor):
+    """Represent an TotalConnect zone low battery status."""
+
+    def update(self):
+        """Return the state of the device."""
+        self._attr_is_on = self._zone.is_low_battery()
+
+
+class TotalConnectTamperBinarySensor(TotalConnectZoneBinarySensor):
+    """Represent an TotalConnect zone tamper status."""
+
+    def update(self):
+        """Return the state of the device."""
+        self._attr_is_on = self._zone.is_tampered()
+
+
+class TotalConnectAlarmBinarySensor(BinarySensorEntity):
+    """Represent an TotalConnect alarm device binary sensors."""
+
+    def __init__(self, location, description):
+        """Initialize the TotalConnect alarm device binary sensor."""
+        self._location = location
+        self.entity_description = description
+        self._attr_unique_id = f"{location.location_id} {description.key}"
+        self._attr_is_on = None
+        self._attr_extra_state_attributes = {
+            "location_id": self._location.location_id,
         }
-        return attributes
+
+
+class TotalConnectAlarmLowBatteryBinarySensor(TotalConnectAlarmBinarySensor):
+    """Represent an TotalConnect Alarm low battery status."""
+
+    def update(self):
+        """Return the state of the device."""
+        self._attr_is_on = self._location.is_low_battery()
```

Comment:
You could add the callable into the entity description and thereby skip having all these classes and just have one.

Reply:
@gjohansson-ST please point me to an example.

Reply:
Here is one homeassistant/components/sensibo/binary_sensor.py

Reply:
Thanks for the example, but I've spent almost two hours trying and I don't understand it well enough to make it work.  

The current method is only a few extra lines of code, and easier for me to understand, so I'd like to keep it as-is.